### PR TITLE
Run golangci-lint as part of pull request workflow #604

### LINKED ISSUE #604: Run golangci-lint as part of pull request workflow
## What

Unfortunately [GolangCI is closing](https://medium.com/golangci/golangci-com-is-closing-d1fc1bd30e0e) and we need to migrate to another alternative. So we can have a sanity check for upcoming pull requests.

## How

* GolangCI service was closed, but the linter [golangci-lint](https://github.com/golangci/golangci-lint) will be maintained, which means that we can benefit from this using GitHub actions.

## Additional information

* [golangci-lint-action](https://github.com/marketplace/actions/run-golangci-lint)

### PR DESCRIPTION
**Pull request by abstractj**
## Changes

- Merge build and linter jobs in a single file
- Runs tests for go 1.13 and 1.14
- Introduces code coverage with coveralls
- Fixes some issues reported by golangci-lint and relax/disable some
rules until the codebase is refactored

## Running example:

- https://github.com/abstractj/louketo-proxy/runs/676376256

## Testing this PR

* Download the `build.yml` file from this PR and add it to `.github/workflows` into the fork of
Louketo
* Push the changes to `master` at your fork
* Submit a pull request to your own fork and GH action should be triggered

Relates to #615 
Resolves #604

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
```
Comment (JoelSpeed) [inline .github/workflows/go.yml:31]: This repo uses go modules right? we can drop this

Comment (abstractj) [reply .github/workflows/go.yml]: Thanks @JoelSpeed updated. I just enabled it on GH and missed that part.

**Comment by abstractj** [comment]
> Is there an easy way we can test these workflows before merging?

Yes, you can look at the fork here https://github.com/abstractj/louketo-proxy/actions/runs/99223707. As you may notice, there are several alerts there for golangci-lint. Which ones do you think we should disable?

IMO the alerts for `whitespace` make sense, but not sure about the `complexity` warnings. Because sometimes we only updated the code to please the Linter. But if you consider this something important, we can keep it.

**Comment by JoelSpeed** [comment]
In general I think the complexity ones are good, but you can get ones that are a bit annoying, such as the ones highlighted here. I believe we can add an exception to `.golangci.yaml` if we want to keep it turned on. I have no strong opinion either way on this one

**Comment by abstractj** [comment]
@JoelSpeed np, let's keep it. Found a typo on my YAML file, but it's fixed now. If there's anything else that you would like to change, please let me know.

**Comment by JoelSpeed** [comment]
Do you think we should try and fixup the linting issues before merging this? Probably as a separate PR?

**Comment by abstractj** [comment]
@JoelSpeed of course, we can do that, there's no need to rush about merging this.

**Comment by abstractj** [comment]
@stianst could you please review?

**Pull request merged**